hack/jenkins/README.md

```diff
@@ -0,0 +1,28 @@
+# Jenkins
+
+[Jenkins](http://jenkins-ci.org/) is a pluggable continuous
+integration system. The Google team is running a Jenkins server on a
+private GCE instance for the Kubernetes project in order to run longer
+integration tests, continuously, on different providers. Currently, we
+(Google) are only running Jenkins on our own providers (GCE and GKE)
+in different flavors.
+
+## General flow
+The flow of the Google Jenkins server:
+* Under the `kubernetes-build` job: Every 5 minutes, Jenkins polls for a batch of new commits, after which it runs the `build.sh` script (in this directory) on the latest tip. This results in build assets getting pushed to GCS and the `latest.txt` file in the `ci` bucket being updated. That job then triggers `kubernetes-e2e-*`.
+* On trigger, and every half hour (which effectively means all the time, unless we're failing cluster creation), e2e variants run, on the latest build assets in GCS:
+  * `kubernetes-e2e-gce`: Standard GCE e2e
+  * `kubernetes-e2e-gke`: GKE provider e2e, with head k8s and GKE creating clusters at its default version
+  * `kubernetes-e2e-gke-ci`: GKE provider e2e, with head k8s and GKE creating clusters at the head k8s version
+* Each named job may run in parallel with the others, i.e. `kubernetes-build` with `kubernetes-e2e-gce`, but `kubernetes-e2e-gce` will not run concurrently with itself.
```

Comment:
You're actually picking on Jenkins configuration here, and if I describe it much more, the referential integrity of this document will drift. I'd rather rip this section out again because it's more or less irrelevant. So I will do so.